### PR TITLE
Integrate ESlint to NodeJS API

### PR DESCRIPTION
**Pull request by njmulsqb**
The PR:

- add eslint as dependency and updates scripts
- run eslint on current code and fixes errors
- include github workflow to add eslint to CI

closes zaproxy/zaproxy#7646

### CODE REVIEW

**Comment by thc202** [comment]
This should be rebased.

**Comment by njmulsqb** [comment]
> This should be rebased.

What will be the command :P

**Comment by thc202** [comment]
Pushed directly to try start CI but didn't work…

**Comment by njmulsqb** [comment]
What should be the solution now?

**Comment by thc202** [comment]
We will have to merge the PR to work.

**Comment by njmulsqb** [comment]
Which PR need to be merged?

**Comment by thc202** [comment]
This PR.

File: package.json [hunk 1]
```diff
@@ -24,6 +24,14 @@
...
     "node": ">=17.0.0"
   },
+  "scripts": {
+    "lint": "eslint ./src",
+    "lint:fix": "eslint --fix ./src"
+  },
+  "scripts": {
+    "lint": "eslint ./src",
+    "lint:fix": "eslint --fix ./src"
+  },
```
Comment (thc202) [inline package.json:34]: This seems to be duplicated.

File: src/accessControl.js [hunk 2]
```diff
@@ -17,32 +17,31 @@
...
 }
 
 /**
  * Gets the Access Control scan progress (percentage integer) for the given context ID.
  * This component is optional and therefore the API will only work if it is installed
  * @param {string} contextid
  **/
-AccessControl.prototype.getScanProgress = function (args) {
-    return this.api.request('/accessControl/view/getScanProgress/', {'contextId': args.contextid })
+AccessControl.prototype.getScanProgress = async function (args) {
```
Comment (thc202) [inline src/accessControl.js:34]: It's adding the `async`/`await`.

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+on:
+  pull_request:
+
+jobs:
+  eslint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (thc202) [inline .github/workflows/lint.yml:11]: Better use latest version, `v3`.

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+name: lint
+
+on:
+  pull_request:
```
Comment (thc202) [inline .github/workflows/lint.yml:4]: Would be good to run this on the default branch as well.

File: .github/workflows/lint.yml
Comment (thc202) [inline .github/workflows/lint.yml:1]: Call it `ci.yml` and `CI` for the workflow name.

File: .github/workflows/lint.yml (hunk 3 above)
Comment (thc202) [reply .github/workflows/lint.yml:11]: Also, add node setup to not rely on default versions.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**